include/grpcpp/ext/orca_service.h

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#ifndef GRPCPP_EXT_ORCA_SERVICE_H
+#define GRPCPP_EXT_ORCA_SERVICE_H
+
+#include <map>
+#include <string>
+
+#include "absl/types/optional.h"
+
+#include <grpcpp/impl/codegen/server_callback.h>
+#include <grpcpp/impl/codegen/service_type.h>
+#include <grpcpp/impl/codegen/sync.h>
+#include <grpcpp/server_builder.h>
+#include <grpcpp/support/slice.h>
+
+namespace grpc {
+namespace experimental {
+
+// RPC service implementation for supplying out-of-band backend
+// utilization metrics to clients.
+class OrcaService : public Service {
+ public:
+  struct Options {
+    // Minimum report interval.  If a client requests an interval lower
+    // than this value, this value will be used instead.
+    int min_report_duration_ms = 30 * 1000;
+
+    Options() = default;
+    Options& set_min_report_duration_ms(int ms) {
```

Comment:
Quick check on whether we can use absl::Duration

Reply:
I had assumed that we didn't want to use this yet based on past discussions, but maybe this is an edge case.  I've started an internal discussion about this, and we'll see what the consensus is.

I don't think this needs to block this PR, because the API is still marked as experimental, so I can change this if we decide to use `absl::Duration`.

Reply:
I've changed it to use `absl::Duration`, as per the result of the internal discussion.